### ISSUE TITLE
Multiple Targets

### ISSUE THREAD

**Issue by mijosch** [jdb78/pytorch-forecasting#670]
There are multiple issues when passing a list of targets.

The generator returns data in a format that the library can not work on.
So if I pass 100 targets, the generator returns a list of 100 times [1,min_encoder_length].
So the first error is, that a list is passed and not a tensor.
Changing that so it gives the right result, I get issues with the target_scale in the Metrics rescale_parameters function.

I try to fix stuff step by step.

Someone already has a solution for this?

**Comment by mijosch** [comment]
little addition, this is how I create the dataset:

training = TimeSeriesDataSet(
    data[lambda x: x.time_idx <= training_cutoff],
    time_idx="time_idx",
    target=list(columns),
    group_ids=["group"],
    min_encoder_length=max_encoder_length // 2,  # keep encoder length long (as it is in the validation set)
    max_encoder_length=max_encoder_length,
    min_prediction_length=1,
    max_prediction_length=max_prediction_length,
    static_categoricals=[],
    static_reals=[],
    time_varying_known_categoricals=[],
    #variable_groups=["group"] ,  # group of categorical variables can be treated as one variable
    time_varying_known_reals=["time_idx"],
    #time_varying_unknown_categoricals=[],
    time_varying_unknown_reals=list(columns),
    target_normalizer=GroupNormalizer(groups=["group"], transformation="softplus"),  # use softplus and normalize by group
    add_relative_time_idx=True,
    add_target_scales=True,
    add_encoder_length=True,
    allow_missing_timesteps=True,
)

**Comment by brysab** [comment]
I have some issues too, especially with time_idx giving me a key error:'time_index'. However, if you have a multi target, it's a multivariate dataset you choose use, so change your target normalizer by: (target_normalizer=MultiNormalizer([EncoderNormalizer(), TorchNormalizer()]). It may help you.
